### ISSUE TITLE
matplotlib fails to detect Tk if path is different from Tcl on Cray 

### ISSUE THREAD

**Issue by gppezzi** [hpcugent/easybuild-easyconfigs#2659]
Problem is that setupext.py from matplotlib is guessing Tk path based on Tcl location:
```
# Guess Tk location based on Tcl location
(head, tail) = os.path.split(tcl_lib_dir)
tail = tail.replace('Tcl', 'Tk').replace('tcl', 'tk')
````
If, for example, the Tk easyconfig file adds versionsuffix = '-no-X11', the replace is not working and matplotlib fails to detect Tk.

I removed no-X11 versionsuffix as workaround, but it would be nice to understand if Tk is working correctly and how the detection is working in other builds.

More info here https://github.com/hpcugent/easybuild-easyconfigs/pull/2608

**Comment by gppezzi** [comment]
I did some further testing with Python/2.7.11-foss-2016a, which has Tk/8.6.4-foss-2016a-no-X11 as dependency. 

When building matplotlib (`python setup.py  build`) I get this:
```
OPTIONAL BACKEND EXTENSIONS
               tkagg: no  [The C/C++ header for Tk (tk.h) could not be found.  You may need to install the development package.]
```

Then I've installed Tk/8.6.4-foss-2016a (dropping the version suffix), I re-ran the same command as above (without loading any new module) and I get this:
```
OPTIONAL BACKEND EXTENSIONS
                 tkagg: yes [installing, version 81008]
```

This test was done on a RH6.7, so this is not cray related.  The only difference on cray is that compilation will fail as matplotlib will try harcoded values for tk/tcl version & path (instead of just disabling the extension as on RH6.7)

**Comment by boegel** [comment]
WIth `matplotlib/1.5.1-intel-2016a-Python-2.7.11` on SL6, which depends on `Tk/8.6.4-intel-2016a-no-X11`, I'm not seeing this problem:

```
tkagg: yes [installing, version 81008] 
```

**Comment by gppezzi** [comment]
Could be related to the pkg-config settings? anyway, it is most probably system dependent. 

How do you know it is finding the tk you have loaded or if it is using the system one?

**Comment by dominikborkowski** [comment]
Same problem here, foss-2016a on SLES11 SP3. Trying to compile matplotlib 1.4.3 and 1.5.1 results in the same error:

/apps/easybuild/software/shadowfax-westmere/binutils/2.25-GCCcore-4.9.3/bin/ld.gold: error: cannot find -ltcl
/apps/easybuild/software/shadowfax-westmere/binutils/2.25-GCCcore-4.9.3/bin/ld.gold: error: cannot find -ltk

After compiling a new Tk without the version suffix (Tk/8.6.4-foss-2016a), matplotlib was able to compile.

**Comment by boegel** [comment]
fixed in #2971 

I'm pretty sure the fix works, so I'll close this, but please confirm @gppezzi and/or @dominikborkowski
